LogfileIterator/PolySyncLogfile.cpp

```diff
@@ -0,0 +1,501 @@
+}
+
+
+void LogfileAttributes::setFileSize(ulong fileSize)
+{
+    _logfileAttributes->file_size = fileSize;
+}
+
+
+void LogfileAttributes::setDataOffset(ulong dataOffset)
+{
+    _logfileAttributes->data_offset = dataOffset;
+}
+
+
+void LogfileAttributes::setDataCount(ulong dataCount)
+{
+    _logfileAttributes->data_count = dataCount;
+}
+
+
+void LogfileAttributes::setAvgDataSize(ulong avgDataSize)
+{
+    _logfileAttributes->avg_data_size = avgDataSize;
+}
+
+
+ps_rnr_log_header LogfileAttributes::getLogHeader() const
+{
+    return _logfileAttributes->log_header;
+}
+
+
+ps_timestamp LogfileAttributes::getStartTime() const
+{
+    return _logfileAttributes->start_time;
+}
+
+
+ps_timestamp LogfileAttributes::getEndTime() const
+{
+    return _logfileAttributes->end_time;
+}
+
+
+ps_timestamp LogfileAttributes::getDuration() const
+{
+    return _logfileAttributes->duration;
+}
+
+
+ulong LogfileAttributes::getFileSize() const
+{
+    return _logfileAttributes->file_size;
+}
+
+
+ulong LogfileAttributes::getDataOffset() const
+{
+    return _logfileAttributes->data_offset;
+}
+
+
+ulong LogfileAttributes::getDataCount() const
+{
+    return _logfileAttributes->data_count;
+}
+
+
+ulong LogfileAttributes::getAvgDataSize() const
+{
+    return _logfileAttributes->avg_data_size;
+}
+
+
+ps_logfile_attributes * LogfileAttributes::getFileAttributes() const
+{
+    return _logfileAttributes;
+}
+
+
+RnrLogRecord::RnrLogRecord()
+{}
+
+
+RnrLogRecord::RnrLogRecord( const ps_rnr_log_record & )
+{}
+
+
+RnrLogRecord::RnrLogRecord( const RnrLogRecord & )
+{}
+
+
+RnrLogRecord::~RnrLogRecord()
+{}
+
+unsigned int RnrLogRecord::getIndex()
+{
+    return _rnrLogRecord->index;
+}
+
+
+unsigned int RnrLogRecord::getSize()
+{
+    return _rnrLogRecord->size;
+}
+
+unsigned int RnrLogRecord::getPreviousSize()
+{
+    return _rnrLogRecord->prev_size;
+}
+
+
+unsigned int RnrLogRecord::getTimestamp()
+{
+    return _rnrLogRecord->timestamp;
+}
+
+
+ps_rnr_log_record * RnrLogRecord::getRnrLogRecord() const
+{
+    return _rnrLogRecord;
+}
+
+
+Logfile::Logfile ( Node & node ) : _node ( node.getNodeReference() )
+{
+    // initialize a node's logfile API resources
+      checkDTC( psync_logfile_init( _node ),
+                  "Failed to initialize LogFile Class." );
+}
+
+
+Logfile::~Logfile()
+{
+    // releasing a node's logfile API resources
+    checkDTC( psync_logfile_release( _node ),
+                "Failed to release node's logfile resources." );
+}
+
+
+void Logfile::setModeRead()
+{
+    setMode( LOGFILE_MODE_READ, _sessionId );
+}
+
+
+void Logfile::setModeWrite()
+{
+    setMode( LOGFILE_MODE_WRITE, _sessionId );
+}
+
+
+void Logfile::setModeOff()
+{
+    setMode( LOGFILE_MODE_OFF, _sessionId );
+}
+
+
+void Logfile::setStateEnabled( ps_timestamp offsetToStart )
+{
+    setState( LOGFILE_STATE_ENABLED, offsetToStart );
+}
+
+
+void Logfile::setStateDisabled()
+{
+    setState( LOGFILE_STATE_DISABLED, 1000 );
```

Comment:
The Logfile documentation for the API can be pretty lengthy. Part of that documentation is as such: when enabling state for single node context, timing delay is zero. When enabling state for distributed context, timing delay is now + 2 seconds, generally.

When disabling state and/or setting state to invalid, a near-zero time delay is preferable. This is  a near-zero time delay here that's the "magic number." This is documented in the Logfile API. Near-zero is preferable over zero for timing skew and timing delay.